faux-mgs/src/main.rs

```diff
@@ -1387,6 +1436,96 @@ async fn run_command(
     }
 }
 
+async fn monorail_unlock(
+    log: &Logger,
+    sp: &SingleSp,
+    time_sec: u32,
+    key: Option<PathBuf>,
+) -> Result<()> {
+    let r = sp
+        .component_action_with_response(
+            SpComponent::MONORAIL,
+            ComponentAction::Monorail(
+                MonorailComponentAction::RequestChallenge,
+            ),
+        )
+        .await?;
+
+    let ComponentActionResponse::Monorail(
+        MonorailComponentActionResponse::RequestChallenge(challenge),
+    ) = r
+    else {
+        bail!("unexpected response: {r:?}");
+    };
+    info!(log, "received challenge {challenge:?}");
+
+    let response = match challenge {
+        UnlockChallenge::EcdsaSha2Nistp256(data) => {
+            let Some(priv_path) = key else {
+                bail!("need --key for ECDSA challenge");
+            };
+
+            debug!(log, "loading private key from {priv_path:?}");
+            let priv_key = ssh_key::PrivateKey::read_openssh_file(&priv_path)
```

Comment:
I don't think this works with a yubikey, right? There's no way to read the private key, and it can only be interacted with via ssh-agent.

I'd be fine landing this as-is and then doing the ssh-agent work as a followup, but just to think out loud about that work for a sec:

* Should we talk directly to ssh-agent via something like https://crates.io/crates/ssh-agent-client-rs, or should we shell out to `ssh-keygen`? I'm tempted to lean toward the former, although that crate in particular might need some minor work [to support sha256 in its API](https://docs.rs/ssh-agent-client-rs/0.9.1/src/ssh_agent_client_rs/codec.rs.html#91-95).
* Once we add ssh-agent support, do we need to keep raw private key file support? I'd lean toward "no" but could be convinced otherwise since it wouldn't be hard to keep around.